tests/test_client_functional.py

```diff
@@ -2453,6 +2458,53 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "invalid_redirect_url, error_message_url, expected_exception_class",
+    (
+        # yarl.URL.__new__ raises ValueError
+        ("http://:/", "http://:/", InvalidUrlRedirectClientError),
+        (
+            "http://example.org:non_int_port/",
+            "http://example.org:non_int_port/",
+            InvalidUrlRedirectClientError,
+        ),
+        # # yarl.URL.origin raises ValueError
+        ("http:/", "http:///", InvalidUrlRedirectClientError),
+        ("http:/example.com", "http:///example.com", InvalidUrlRedirectClientError),
+        ("http:///example.com", "http:///example.com", InvalidUrlRedirectClientError),
+        ("bluesky://d:i:d", "bluesky://d:i:d", InvalidUrlRedirectClientError),
```

Comment:
This URL is valid. It's just non-HTTP.

Reply:
It's result of yarl's ValueError here:

```
try:
    parsed_redirect_url = URL(
        r_url, encoded=not self._requote_redirect_url
    )
except ValueError as e:
    raise InvalidUrlRedirectClientError(
        r_url,
        "Server attempted redirecting to a location that does not look like a URL",
    ) from e
```

```
...
>               raise ValueError(message) from None
E               ValueError: Port could not be cast to integer value as 'i:d'
```

it seems to be invalid

Reply:
@setla you're right. I came up an incorrect URL example. I suggested changes here: https://github.com/aio-libs/aiohttp/pull/6722/files#r1485732854.